packages/indy-sdk-to-askar-migration/src/IndySdkToAskarMigrationUpdater.ts

```diff
@@ -0,0 +1,411 @@
+
+    // Create the directories for the backup
+    await this.fs.createDirectory(dest)
+
+    // Copy the supplied database to the backup destination
+    await this.fs.copyFile(src, dest)
+
+    if (!(await this.fs.exists(dest))) {
+      throw new IndySdkToAskarMigrationError('Could not locate the new backup file')
+    }
+  }
+
+  /**
+   * Reverts backed up database file to the original path, if its missing, and
+   * deletes the backup. We do some additional, possible redundant, exists checks
+   * here to be extra sure that only a happy flow occurs.
+   */
+  private async restoreDatabase() {
+    // "Impossible" state. Since we do not continue if `this.backupDatabase()`
+    // fails, this file should always be there. If this error is thrown, we
+    // cannot correctly restore the state.
+    if (!(await this.fs.exists(this.backupFile))) {
+      throw new IndySdkToAskarMigrationError('Backup file could not be found while trying to restore the state')
+    }
+
+    /**
+     * Since we used `copy` to get the file, it should still be there. We
+     * double-check here to be sure.
+     */
+    if (!(await this.fs.exists(this.dbPath))) {
+      return
+    } else {
+      this.agent.config.logger.trace(`Moving '${this.backupFile}' back to the original path: '${this.dbPath}`)
+
+      // Move the backedup file back to the original path
+      await this.fs.copyFile(this.backupFile, this.dbPath)
+
+      this.agent.config.logger.trace(`Cleaned up the backed up file at '${this.backupFile}'`)
+    }
+  }
+
+  // Delete the backup as `this.fs.copyFile` only copies and no deletion
+  // Since we use `tempPath` which is cleared when certain events happen,
+  // e.g. cron-job and system restart (depending on the os) we could omit
+  // this call `await this.fs.delete(this.backupFile)`.
+  private async cleanBackup() {
+    this.agent.config.logger.trace(`Deleting the backup file at '${this.backupFile}'`)
+    await this.fs.delete(this.backupFile)
+  }
+
+  /**
+   * Move the migrated and updated database file to the new location according
+   * to the `FileSystem.dataPath`.
+   */
+  private async moveToNewLocation() {
+    const src = this.dbPath
+    // New path for the database
+    const dest = this.newWalletPath
+
+    // create the wallet directory
+    await this.fs.createDirectory(dest)
+
+    this.agent.config.logger.trace(`Moving upgraded database from ${src} to ${dest}`)
+
+    // Copy the file from the database path to the new location
+    await this.fs.copyFile(src, dest)
+
+    // Delete the original, only if specified by the user
+    if (this.deleteOnFinish) await this.fs.delete(this.dbPath)
+  }
+
+  /**
+   * Function that updates the values from an indy-sdk structure to the new askar stucture.
+   *
+   * - Assert that the paths that will be used are free
+   * - Create a backup of the database
+   * - Migrate the database to askar structure
+   * - Update the Keys
+   * - Update the Master Secret (Link Sercet)
```

Comment:
```suggestion
   * - Update the Master Secret (Link Secret)
```

The classic @blu3beri typo!